src/platform/nm-linux-platform.c

```diff
@@ -3574,6 +3582,40 @@ _new_from_nl_qdisc (struct nlmsghdr *nlh, gboolean id_only)
 	obj->qdisc.parent = tcm->tcm_parent;
 	obj->qdisc.info = tcm->tcm_info;
 
+	if (tb[TCA_OPTIONS]) {
+		struct nlattr *options_attr;
+		int remaining;
+
+		nla_for_each_nested (options_attr, tb[TCA_OPTIONS], remaining) {
+			switch (nla_type (options_attr)) {
```

Comment:
I think this lacks validation, doesn't it? You just assume that `options_attr` contains a u32 type.